### DISCUSSION TITLE
Never attach an OS disk/USB to LBB

### DISCUSSION THREAD

**Discussion by sejtam** [outdoorbits/little-backup-box#217] [Q&A]
So I made a mistake. One of my other RPis (headless) had a problem such that i was unable to remotely log in , and having the LBB just set up, i thought

"why don't i just pop its SD card/thumbdrive into the LBB, mount it there and have a look at the logs"

**BIG MISTAKE**

LBB seems to have auto-mounted the drive under /media/storage and done a

   `# chmod -R 755 /media/storage`

(and the same to /media/boot once I mounted /dev/sda1 there to see whether the 'ssh' file needed to be added again to allow remote access)

That all of course royally screwed up the disk, so that almost nothing worked anymore (SSH, cron all complained about too open file permissions)

Would it be possible to have LBB save a file with 'undo' commands when it does something drastic like that

I am thinking of something like

```
# exec 5>lbb_magledperms_undo
# find . -print | sed -e 's|^\.\/||' | while read F; do
  PERMS=$(stat -c %a ${F})
  OWNERS=$(stat -c "%u:%g" %{F})
  echo "chmod ${PERMS} ${F}" >&5
  echo "chown ${OWNERS} ${F}" >&5
  done
```
(note: currently untested)

to be run *before* LBB adjusts the perms recursively

There may be additional attributes that may need resetting !

I am unsure whether that undo file should be saved on the mounted partition itself
or in /tmp or /var/tmp
if in /tmp or /var/tmp it would not be available on the mangled disk itself later (and thus will need the LBB again)
and also it would need to somehow include an indicator of which disk it was applied

so I am leaning more to saving the undo file on the affected disk itself...

By having such an 'undo' file it will be rather trivial to repair the damage done
The file could be overwritten on the net use

**Comment by sejtam** [comment]
I should of course have made a new generic RPI SDcard and used that rather than the LBB, but I forgot

**Comment by outdoorbits** [comment]
I'm actually rewriting all the core code to python. In the upcoming version I will try not to need chmod for the mounted source device anymore. If this won't be possible, maybe you could contribute your idea.

**Reply by outdoorbits** [reply]
The new version won't change permissions anymore.
